namer/consul/src/main/scala/io/buoyant/namer/consul/SvcAddr.scala

```diff
@@ -117,11 +117,11 @@ private[consul] object SvcAddr {
               failureLogLevel,
               "consul datacenter '%s' service '%s' observation error %s." +
                 " Last known state is %s",
-              datacenter, key.name, e, currentValueToLog
+              datacenter, key.name, e, currentState
             )
             val backoff #:: nextBackoffs = backoffs
             // subsequent errors are logged as DEBUG
-            Future.sleep(backoff).before(loop(None, nextBackoffs, Level.DEBUG, currentValueToLog))
+            Future.sleep(backoff).before(loop(None, nextBackoffs, Level.DEBUG, currentState))
 
           case Return(v1.Indexed(_, None)) =>
             // If consul doesn't return an index, we're in bad shape.
```

Comment:
I just checked 2 months of our logs (14 namerd instances talking to 5 datacenters in a pretty flaky consul cluster) and this message hasn't happenned even once.

@adleong , do you happen to recall why was this branch added in the first place? As of today, it looks to me, as if this was effectively dead code.

Reply:
@edio I think you're right.  It looks like these two branches used to do different things (a 5XX would update the state to Neg whereas a timeout would not) but a recent change (https://github.com/linkerd/linkerd/commit/a17406b43a928a59c8d1bdd94ddcb0f0bc8588cb) effectively make these branches identical.  I think we can safely unify them.

Reply:
@adleong , I believe, though, you replied to a wrong thread ;)

Reply:
whoops, you're right.  I don't think this was added for any specific use-case, but just as standard defensive programming.  See: https://github.com/linkerd/linkerd/pull/632/files#diff-ea0487e1d8c6b88af1a70cdbf4dc02f6R55

Reply:
I see. I personally think, this code, does more harm than good. I concluded from the code, that there's some case, where consul behaves like this. And, as I said, in the last 2 months we haven't seen this branch of code executed even once, so I'm almost certain, it is indeed never executed.

I don't have strong opinion on this, but I'd prefer to remove this branch of code. Would you mind if I did that?

Reply:
I see where you're coming from.  This branch also is questionable because it halts the request loop instead of restarting the watch or something like that, which is probably not great.  But still, since it's mostly unrelated to this change, I'd prefer we don't make changes unless we have a clear motivation for doing so.

Reply:
I reverted my changes in that place in the code to keep this PR as relevant to the issue as possible.